src/flp/test_types.rs

```diff
@@ -0,0 +1,112 @@
+use crate::field::FieldElement;
+use crate::flp::{FlpError, Gadget, Type};
+use std::convert::{TryFrom, TryInto};
+
+/// The average type. Each measurement is an integer in `[0,2^bits)` for some `0 < bits < 64` and the
+/// aggregate is the arithmetic average. This type is a test for demonstrating that the number of measurements
+/// may be used in the `decode_result()` function. No validity checks are done on the input data.
+///
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Average<F: FieldElement> {
+    bits: usize,
+    one: F::Integer,
+}
+
+impl<F: FieldElement> Average<F> {
+    pub fn new(bits: usize) -> Result<Self, FlpError> {
+        let bits_int = F::Integer::try_from(bits).map_err(|err| {
+            FlpError::Encode(format!(
+                "bit length ({}) cannot be represented as a field element: {:?}",
+                bits, err,
+            ))
+        })?;
+
+        if F::modulus() >> bits_int == F::Integer::from(F::zero()) {
+            return Err(FlpError::Encode(format!(
+                "bit length ({}) exceeds field modulus",
+                bits,
+            )));
+        }
+
+        let one = F::Integer::from(F::one());
+
+        Ok(Self { bits, one })
+    }
+}
+
+impl<F: FieldElement> Type for Average<F> {
+    type Measurement = F::Integer;
+    type AggregateResult = f64;
+    type Field = F;
+
+    fn encode_measurement(&self, summand: &F::Integer) -> Result<Vec<F>, FlpError> {
```

Comment:
It should be possible to refactor `Average` and `Sum` to reuse code. In particular:
* encode_measurement()
* decode_result()
* truncate()
* valid()

all have lots of code in common. (There might be others, too.)